signer/src/stacks/events.rs

```diff
@@ -23,11 +23,14 @@ use blockstack_lib::burnchains::Txid as StacksTxid;
 use clarity::vm::types::CharType;
 use clarity::vm::types::PrincipalData;
 use clarity::vm::types::SequenceData;
+use clarity::vm::types::SequencedValue;
 use clarity::vm::types::TupleData;
 use clarity::vm::ClarityName;
 use clarity::vm::Value as ClarityValue;
 use stacks_common::types::chainstate::StacksBlockId;
 
+use crate::keys::PublicKey;
```

Comment:
I think we should change this to use the `secp256k1::PublicKey` type instead.

So, this file uses no types from the rest of the signer crate, it only uses rust-bitcoin types, stacks-core types, and bitvec types (and the rust-bitcoin crate has a dependency on the `secp256k1` crate so that is fine). The reason for us limiting the types used here is that we want to make it easy for people to build on sBTC without pulling in all of the dependencies that we have in the signer crate. 

Eventually we should move this module to the sBTC crate, but first we'd have to remove our use of the `blockstack_lib::burnchains::Txid`, which pulls in a lot of stuff. We may also want to remove our use of bitvec too.

Reply:
That's a great point! For now I only refactored with the use of `secp256k1::PublicKey`